src/stash/mod.rs

```diff
@@ -364,20 +364,41 @@ impl<V> Stash<V> {
 
     /// Take an item from a slot (if non empty).
     pub fn take(&mut self, index: usize) -> Option<V> {
-        if let Some(entry) = self.data.get_mut(index) {
-            match mem::replace(entry, Entry::Empty(self.next_free)) {
+        match self.data.get_mut(index) {
```

Comment:
yeah you are right. It was just a test if Rust compiler was able to go for more optimization if I used a single epression for the entire function body. But there were no differences in performance. Besides that it now looks more uniformely since I implemented the other methods in that style, too. Of course, it is your decision if you want to pull it that way.